core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -139,11 +139,14 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
    * we need to add a delay of X to W such that O * W / (W + X) = T.
    * Solving for X, we get X = (O - T)/T * W.
    */
-  private def throttleTime(metricValue: Double, config: MetricConfig): Int = {
+  private def throttleTime(clientMetric: KafkaMetric, config: MetricConfig): Int = {
+    // Casting to Rate because we only use Rate in Quota computation
+    val rateMetric = clientMetric.measurable().asInstanceOf[Rate]
```

Comment:
Thanks for the explanation. My take is that what I suggested is still useful. Why? We are relying on a particular implementation detail in another method that is not documented or enforced in that other method. It seems plausible that someone else may add a different measurable type at some point without knowing that we rely on this implementation, which would then trigger a ClassCastException instead of a more relevant exception with a more helpful message. In Scala, `asInstanceOf` is generally considered bad practice for this reason.

Having said that, usages of `asInstanceOf` like this one are common in Kafka code today and you may think that the scenario I outlined is unlikely based on your knowledge of this code. So, I leave it up to you. :)